dot/network/README.md

```diff
@@ -0,0 +1,178 @@
+[the official `libp2p` adapter](https://github.com/libp2p/go-libp2p-yamux) for Yamux.
+
+#### Kademlia
+
+[Kademlia](https://en.wikipedia.org/wiki/Kademlia) is a battle-tested
+[distributed hash table (DHT)](https://en.wikipedia.org/wiki/Distributed_hash_table) that defines methods for managing a
+dynamic list of peers that is constantly updated in order to make a P2P network more resilient and resistant to attacks.
+Kademlia calculates a logical "distance" between any two nodes in the network by applying the xor operation to the IDs
+of those two peers. Although this "distance" is not correlated to the physical distance between the peers, it adheres to
+three properties that are [crucial to the analysis](https://en.wikipedia.org/wiki/Kademlia#Academic_significance) of
+Kademlia as a protocol - in particular, these three properties are:
+
+- the "distance" between a peer and itself is zero
+- the "distance" between two peers is the same regardless of the order in which the peers are considered (it is
+  [symmetric](https://en.wikipedia.org/wiki/Symmetry_in_mathematics))
+- the shortest "distance" between two peers does not include any intermediate peers (it follows the
+  [triangle inequality](https://en.wikipedia.org/wiki/Triangle_inequality))
+
+Gossamer uses [the official `libp2p` implementation of Kademlia for Go](https://github.com/libp2p/go-libp2p-kad-dht).
+
+### Blockchain-Specific Protocols
+
+The `libp2p` stack is used to implement the blockchain-specific protocols that are used to participate in
+"Substrate-like" networks - these protocols are divided into two types, [notification](#notification-protocols) and
+[request/response](#requestresponse-protocols). The two types of protocols are described in greater details below, along
+with the specific protocols for each type.
+
+##### Notification Protocols
+
+[Notification protocols](https://crates.parity.io/sc_network/index.html#notifications-protocols) allow peers to
+unidirectionally "push" information to other peers in the network. Although the peer receiving the information must
+explicitly accept a handshake in order to open a stream for a notification protocol, this stream does not allow the
+receiver to "push" information to the sender.
+
+###### Transactions
+
+This protocol is used to notify network peers of [transactions](https://docs.substrate.io/v3/concepts/tx-pool/) that
+have been locally received and validated. Transactions are used to access the
+[public APIs of blockchain runtimes](https://docs.substrate.io/v3/concepts/extrinsics/#signed-transactions).
+
+###### Block Announces
+
+The block announce protocol is used to notify network peers of the existence of a block. The message for this protocol
+contains a [block header](https://docs.substrate.io/v3/getting-started/glossary/#header) and associated data, such as
+the [BABE pre-runtime digest](https://crates.parity.io/sp_consensus_babe/digests/enum.PreDigest.html).
```

Comment:
the BABE digests are part of the header, so I'd just leave this as saying the message is a block header.